app/services/fact_importer.rb

```diff
@@ -40,6 +41,7 @@ def import!
     add_new_facts
     update_facts
 
+    raise ::Foreman::Exception.new(N_("Import of facts failed")) if @error
```

Comment:
Just that it's clear from the error message what's wrong, on the other hand, one would not see the error in user interface anyway, so if one would see it in the logs, there is no point in having it in exception. But having a hostname for which it happened would be a valuable information to have in the error message, especially when multiple imports happen at the same time